### ISSUE TITLE
Allow transforms to output more than one record per input record

### ISSUE THREAD

**Issue by davidpablocohn** [OceanDataTools/openrvdas#180]
Right now, a transform can output at zero or one output record per input record. This becomes problematic when, e.g. a GPS sensor output multiple NMEA records in one blast. Some components, like CachedDataReaders can output single records that have information from multiple records by sending "field dicts", dictionaries where each field, instead of having a single value, has a list of (timestamp, value) pairs:

```
       {
         'data_id': ...,
         'timestamp': ...,
         'fields': {
            field_name: [(timestamp, value), (timestamp, value),...],
            field_name: [(timestamp, value), (timestamp, value),...],
            ...
         }
       }
```

This is computationally efficient, but clunky from an algorithmic standpoint. It would be much cleaner if, instead, they were to output a list of individual records.

**Comment by davidpablocohn** [comment]
The one thing left to do here is to alter a couple of the readers (Serial and Network) so that if they get more than one record per call - as defined by their eol character - they break the input up and return them all as a list of records.

**Comment by davidpablocohn** [comment]
Resolved by 86ad15f

Leaving SerialReader as is (for now).
